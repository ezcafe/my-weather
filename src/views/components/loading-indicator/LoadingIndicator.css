.ai.spinner-border {
    display: block;
    margin: 10px auto;
}
