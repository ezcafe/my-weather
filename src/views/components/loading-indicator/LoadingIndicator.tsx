import { Spinner } from 'react-bootstrap';

import './LoadingIndicator.css';

interface IProps {
    readonly message?: string;
}

const LoadingIndicator: React.FC<IProps> = ({
    message = 'Loading...',
}: IProps) => (
    <Spinner animation="border" role="status" size="sm" className="ai">
        <span className="sr-only">{message}</span>
    </Spinner>
);

export default LoadingIndicator;
