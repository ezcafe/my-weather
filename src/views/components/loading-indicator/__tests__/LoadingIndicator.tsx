import { render } from '@testing-library/react';
import LoadingIndicator from '../LoadingIndicator';

describe('LoadingIndicator', () => {
    test('should show component', async () => {
        const { container } = render(<LoadingIndicator message="Hi there" />);

        expect(container).toMatchSnapshot();
    });
});
