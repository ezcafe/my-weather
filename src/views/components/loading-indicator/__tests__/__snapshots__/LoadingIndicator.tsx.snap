// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator should show component 1`] = `
<div>
  <div
    class="ai spinner-border spinner-border-sm"
    role="status"
  >
    <span
      class="sr-only"
    >
      Hi there
    </span>
  </div>
</div>
`;
